pkg/check/pingpong/dynamic.go

```diff
@@ -0,0 +1,116 @@
+package pingpong
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/random"
+)
+
+// Actions for dynamic behavior
+type Actions struct {
+	NodeGroup   string
+	AddCount    int
+	StartCount  int
+	StopCount   int
+	DeleteCount int
+}
+
+// CheckDynamic executes PingPong check on dynamic cluster
+func CheckDynamic(ctx context.Context, cluster *bee.Cluster, o Options) (err error) {
+	rnd := random.PseudoGenerator((o.Seed))
+	fmt.Printf("Seed: %d\n", o.Seed)
+
+	fmt.Println("Checking PingPong")
+	if err := Check(ctx, cluster, o); err != nil {
+		return fmt.Errorf("check pingpong: %w", err)
+	}
+
+	overlays, err := cluster.Overlays(ctx)
```

Comment:
Can this Overlays call be inside the for loop at start, avoiding the one at the end of the loop?

Reply:
I restructured it. Thanks